eth/taiko_api_backend.go

```diff
@@ -109,3 +118,50 @@ func (s *TaikoAPIBackend) TxPoolContent(
 
 	return txLists, nil
 }
+
+func filterInexecutableTxs(
+	ctx context.Context,
+	backend ethapi.Backend,
+	pendings map[common.Address]types.Transactions,
+) (map[common.Address]types.Transactions, error) {
+	executableTxs := make(map[common.Address]types.Transactions)
+	currentHead := rpc.BlockNumber(backend.CurrentHeader().Number.Int64())
+	// Resolve block number and use its state to ask for the nonce
+	state, _, err := backend.StateAndHeaderByNumberOrHash(ctx, rpc.BlockNumberOrHash{
+		BlockNumber: &currentHead,
+	})
+	if state == nil || err != nil {
+		return nil, err
+	}
+
+	for addr, txs := range pendings {
+		pendingTxs := make(types.Transactions, 0)
+		for i, tx := range txs {
+			// Check that account's nonce at first
+			if i == 0 {
+				nonce := state.GetNonce(addr)
+				if tx.Nonce() != nonce {
+					log.Debug(
+						"Skip a transaction with an invalid nonce",
```

Comment:
Hi @davidtaikocha , This feature is quite cool, but haven't we already implemented (st *StateTransition) preCheck()? It seems to redundantly re-implement StateAndHeaderByNumberOrHash and GetNonce. In the context of txpool, which is a high-frequency scenario, could this add unnecessary pressure on performance? Additionally, there is another issue: after this implementation, would it mean that users can only send one transaction per block? For example, if user A sends three transactions simultaneously with nonces 1, 2, and 3, in Geth, it's possible for these three transactions to be packaged in the same block if they are sorted initially. But with this current feature, transactions with nonces 2 and 3 would absolutely not be included in the latest block. Please correct me if I'm misunderstanding.